src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -364,11 +364,11 @@ int xocl_hot_reset(struct xocl_dev *xdev, u32 flag)
 		if (flag & XOCL_RESET_NO)
 			return 0;
 
-		xocl_peer_request(xdev, &mbreq, sizeof(struct xcl_mailbox_req),
+		mbret = xocl_peer_request(xdev, &mbreq, sizeof(struct xcl_mailbox_req),
 			&ret, &resplen, NULL, NULL, 0, 6);
 		/* userpf will back online till receiving mgmtpf notification */
 
-		return 0;
+		return mbret;
```

Comment:
this is not correct. 
both mbret and ret should be checked. and ret of ESHUTDOWN is the errno returned in mbx response when reset is issued successfully.
pls refer to the code in same function.
also pls try to make the code in this function neat -- current code looks really messy.

Reply:
Hi @xuhz , when I checked both mbret and ret values, I see only mbret has some error code when hot reset request is getting timed out.

> [  +0.000380] xocl 0101:00:00.0: mailbox.u.9437184 ffff92bbde687810 mailbox_request: sending request: 5 via SW
> [  +0.012797] xocl 0101:00:00.0: mailbox.u.9437184 ffff92bbde687810 mailbox_read: Software TX msg is too big
> [  +0.950709] xocl 0101:00:00.0: mailbox.u.9437184 ffff92bbde687810 xocl_drvinst_close: CLOSE 2
> [  +0.000007] xocl 0101:00:00.0: mailbox.u.9437184 ffff92bbde687810 xocl_drvinst_close: NOTIFY 00000000f61ca0e9
> [Feb17 22:07] xocl 0101:00:00.0: mailbox.u.9437184 ffff92bbde687810 timeout_msg: found waiting msg time'd out
> [  +0.000147] xocl 0101:00:00.0:  ffff92ba3319a0b0 xocl_hot_reset: +++mbret: -62, ret: 0
> [  +0.000007] xocl 0101:00:00.0:  ffff92ba3319a0b0 xocl_work_cb: reset failed with error -62

Reply:
if the mailbox msg doesn't time out, eg. the S/N missing case, the mpd plugin will return -2040 that would be in ret (response)

Reply:
and if the reset is sent to wireserver, the mailbox msg response will be ESHUTDOWN (in ret too), which tells the xocl that the whole xocl would be gone. In this case, reset will be waiting in userland till the xocl probe set the dev online

Reply:
yes, I will add code to check ret value also.